s3/s3.go

```diff
@@ -77,19 +77,20 @@ const (
 )
 
 type S3ClientOpts struct {
-	Endpoint        string
-	AddressingStyle AddressingStyle
-	Region          string
-	Secure          bool
-	Transport       http.RoundTripper
-	AccessKey       string
-	SecretKey       string
-	SessionToken    string
-	Trace           bool
-	RoleARN         string
-	RoleSessionName string
-	UseSDKCreds     bool
-	EncryptOpts     EncryptOpts
+	Endpoint         string
+	AddressingStyle  AddressingStyle
+	Region           string
+	Secure           bool
+	Transport        http.RoundTripper
+	AccessKey        string
+	SecretKey        string
+	SessionToken     string
+	Trace            bool
+	RoleARN          string
+	RoleSessionName  string
+	UseSDKCreds      bool
+	EncryptOpts      EncryptOpts
+	DisableDualStack bool
```

Comment:
I am wondering if we should just inherit options here and parse a JSON blob in Argo Workflows's configuration of `extraOpts`. As this continues to expand and add to the spec. Allowing arbitrary options may make more sense.

Reply:
you mean like this?
```go
type ExtraOpts struct {
	EnableDualStack bool
}

type S3ClientOpts struct {
        ...
	ExtraOpts       ExtraOpts
}
```